tutorials/connect-jamf-pro-to-smallstep.mdx

```diff
@@ -0,0 +1,195 @@
+---
+title: Connect Jamf Pro to Smallstep
+html_title: Connect Jamf Pro to Smallstep
+description: Configure Jamf Pro to deploy the Smallstep Agent and distribute certificates and configuration to Mac clients.
+---
+
+Smallstep can integrate with Jamf Pro to synchronize your device inventory, and enroll your fleet with Smallstep using the Smallstep Agent. In this document, we will configure your Jamf Pro instance for use with your Smallstep team.
+
+# Prerequisites
+
+You will need:
+
+- A [Smallstep team](https://smallstep.com/signup)
+- A [Jamf Pro](https://www.jamf.com/products/jamf-pro/) tenant
+- A [package distribution point](https://docs.jamf.com/10.35.0/jamf-pro/install-guide-linux/About_Distribution_Points.html) configured in Jamf Pro
+
+# Step-by-step instructions
+
+### 1. Create API Client in Jamf Pro
+
+This API client will allow Smallstep to read your Jamf device inventory and manage the webhooks necessary for ongoing inventory syncing.
+
+1. In Jamf Pro, choose ⚙️ **Settings**
+2. Under the **System** tab, choose **API Roles and Clients**
+3. Add an **API Role**
+    
+    Display name: Smallstep
+    
+    For Privileges, choose:
+    
+    - Read Mobile Devices
+    - Read Computers
+    - Webhooks: Create, Delete, Read, Update
+4. Next, in the **API Clients** tab, add a **Client**
+    - Display name: Smallstep
+    - API Roles: Smallstep
+    - Enabled: Yes, choose Enable
+    - Choose **Save**
+5. Choose **Generate client secret**
+6. Temporarily save the **Client ID** and **Client Secret**. You’ll use them in the next step.
+
+### 2. Connect Jamf to Smallstep
+
+Let’s add the Jamf API credentials you just created to Smallstep.
+
+1. In the Smallstep UI, go to the [**Device Management**](https://smallstep.com/app/?next=/settings/devices) tab in ⛭ **Settings**
+2. Under Jamf, choose ➕ **Connect**
+3. Enter the API credentials from Jamf
+4. Choose **Add Platform.** Your device inventory will start syncing from Jamf to Smallstep.
+5. Temporarily save the **Webhook URL**, **Username**, and **Password** presented.
+
+### 3. Configure a SCEP Enrollment webhook in Jamf Pro
+
+Smallstep uses Dynamic SCEP certificate enrollment. This webhook allows Jamf to request a dynamic, short-lived challenge string from Smallstep during MDM enrollment. The challenge string is injected into a [SCEP payload](https://developer.apple.com/documentation/devicemanagement/scep) for your clients to authenticate to Smallstep.
+
+1. In Jamf Pro, choose ⚙️ **Settings**
```

Comment:
While technically possible to use our CA for this purpose (caveat below), the certificates that are issued are not MDM enrollment certificates. Users of Jamf will most likely still use the built-in Jamf CA for the MDM enrollment certificates. Calling it certificate enrollment in this context could thus be confusing.

To make it actually work for MDM enrollment certificates with the new Device Identity product, devices may need to be preregistered in another manner. Because if they're not yet enrolled in Jamf, they're not synced over, and they might be blocked from getting a certificate. Haven't tested this flow yet, though.